tests/test_auth.py

```
@@ -0,0 +1,221 @@
+        assert data["is_admin"] is False
+        assert "id" in data
+
+        # Verify user was created in database
+        user = db.query(User).filter(User.username == "newuser").first()
+        assert user is not None
+        assert user.email == "newuser@example.com"
+        assert user.password_hash is not None
+        assert user.password_hash != "securepassword123"  # Should be hashed
+
+    @pytest.mark.asyncio
+    async def test_register_user_duplicate_username(self, client: AsyncClient, db: Session) -> None:
```

Ruff (0.14.13)

Unused method argument: `db`

(ARG002)